src/App.tsx

```diff
@@ -26,70 +26,127 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  switch (sortType) {
+    case SortType.LENGTH:
+      visibleGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    case SortType.ALPABET:
+      visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  sortAlphabet = () => this.setState(() => ({
+    sortType: SortType.ALPABET,
+  }));
+
+  sortByLength = () => this.setState(() => ({
+    sortType: SortType.LENGTH,
+  }));
+
+  getReverse = () => this.setState((state: State) => ({
+    isReversed: !state.isReversed,
+  }));
+
+  getReset = () => this.setState(() => ({
+    sortType: SortType.NONE,
+    isReversed: false,
+  }));
+
+  render(): ReactNode {
+    const { isReversed, sortType } = this.state;
+    const {
+      sortAlphabet,
+      sortByLength,
+      getReverse,
+      getReset,
+    } = this;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={(
+              sortType === SortType.ALPABET
+                ? 'button is-info'
+                : 'button is-info is-light'
+            )}
```

Comment:
Install [classnames](https://www.npmjs.com/package/classnames) and use it instead of ternary operator.
The ternary operator is very limited in its functionality to be used in classNames.

Do the same all over the code.